esmvaltool/preprocessor/_area.py

```diff
@@ -109,21 +131,12 @@ def zonal_means(cube, coordinate, mean_type):
     iris.cube.Cube
         Returns a cube
     """
-    if mean_type == 'mean':
-        result = cube.collapsed(coordinate, iris.analysis.MEAN)
-    elif mean_type == 'stdev':
-        result = cube.collapsed(coordinate, iris.analysis.STD_DEV)
-    elif mean_type == 'variance':
-        result = cube.collapsed(coordinate, iris.analysis.VARIANCE)
-    elif mean_type.lower() in ['minimum', 'min']:
-        result = cube.collapsed(coordinate, iris.analysis.MIN)
-    elif mean_type.lower() in ['maximum', 'max']:
-        result = cube.collapsed(coordinate, iris.analysis.MAX)
-    return result
+    operation = get_iris_analysis_operation(mean_type)
+    return cube.collapsed(coordinate, operation)
 
 
 # get the area average
-def average_region(cube, coord1, coord2, fx_files=None):
+def average_region(cube, coord1, coord2, operator='mean', fx_files=None):
```

Comment:
With the new features added, this function has become too complex according to Codacy. Can you try making it simpler?

Reply:
I've moved the tiling into it's own function now. Should reduce the complexity. I'm not sure whether it will be enough for codacy though.